### DISCUSSION TITLE
[Feature Request]: PWM Fan control?

### DISCUSSION THREAD

**Discussion by gjelsoe** [meshtastic/firmware#4815] [Ideas]
### Platform

other

### Description

Would it be an idea to add PWM control of a fan ?

Devices from Radiomaster like Bandit and Bandit Nano do have a fan connected to the ESP32
and at the moment it can be turned on or off.

So a PWM could be a middle ground if full speed is not necessary but some cooling is wanted.

A setting with LoRa config (FAN speed) could be added where a value between 0-100 (%) can be set.

**Comment by caveman99** [comment]
This seems like an awful lot of work (and additional settings) outside of the core application for meshtastic. Also the Bandit devices use the older SX1276 chip which doesn't make it a Tier1 device for support.

**Reply by gjelsoe** [reply]
I've already done most of the work and it only requires one change to protobufs so a value from 0 to 100 in order to set duty cycle.

**Reply by caveman99** [reply]
For backwards compatibility you can't change protobufs, just add to them. My point being, we're trying to reduce the number of settings, nut inflate them :-)

**Reply by gjelsoe** [reply]
Wasn't going to change one that would break compatibility.
